PlPlot with WX driver (backend) doesn't build on Darwin with error message

```
[ 24%] Building CXX object src/CMakeFiles/plplot.dir/__/drivers/wxwidgets.cpp.o
/tmp/nix-build-plplot-5.15.0.drv-0/plplot-5.15.0/drivers/wxwidgets.cpp:46:18: fatal error: 'Carbon/Carbon.h' file not found
        #include <Carbon/Carbon.h>
                 ^~~~~~~~~~~~~~~~~
1 error generated.
make[2]: *** [src/CMakeFiles/plplot.dir/build.make:636: src/CMakeFiles/plplot.dir/__/drivers/wxwidgets.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1930: src/CMakeFiles/plplot.dir/all] Error 2
make: *** [Makefile:166: all] Error 2
builder for '/nix/store/shik5d6x9rpnkml3pyj8p5h4zh5kmfz8-plplot-5.15.0.drv' failed with exit code 2
error: build of '/nix/store/shik5d6x9rpnkml3pyj8p5h4zh5kmfz8-plplot-5.15.0.drv' failed
```

It might be just a missing dependency to `graphite2`, it would be better to get fixed in the expression of `plplot` or `wxmac` than hacking here.

--- a/pkgs/development/libraries/plplot/default.nix
+++ b/pkgs/development/libraries/plplot/default.nix
@@ -1,6 +1,16 @@
-{ lib, stdenv, fetchurl, cmake }:
+{ lib
+, stdenv
+, fetchurl
+, cmake
+, enableWX ? false
+, wxGTK31, wxmac
+, enableXWin ? false
+, libX11
+}:
 
-stdenv.mkDerivation rec {
+let
+  wxWidgets = (if stdenv.isDarwin then wxmac else wxGTK31);
+in stdenv.mkDerivation rec {
   pname   = "plplot";
   version = "5.15.0";
 
@@ -10,6 +20,21 @@ stdenv.mkDerivation rec {
   };
 
   nativeBuildInputs = [ cmake ];
+
+  buildInputs = [
+  ]
+  ++ lib.optional enableWX wxWidgets
+  ++ lib.optional enableXWin libX11
+  ;
+
+  passthru = {
+    inherit
+      enableWX
+      wxWidgets
+      enableXWin
+      libX11
+    ;
+  };
 
   cmakeFlags = [ "-DCMAKE_SKIP_BUILD_RPATH=OFF" "-DBUILD_TEST=ON" ];
 

--- a/pkgs/tools/misc/hdf4/default.nix
+++ b/pkgs/tools/misc/hdf4/default.nix
@@ -7,7 +7,8 @@
 , uselibtirpc ? stdenv.isLinux
 , libtirpc
 , zlib
-, szip ? null
+, szipSupport ? false
+, szip
 , javaSupport ? false
 , jdk
 }:
@@ -55,10 +56,10 @@ stdenv.mkDerivation rec {
 
   buildInputs = [
     libjpeg
-    szip
     zlib
   ]
   ++ lib.optional javaSupport jdk
+  ++ lib.optional szipSupport szip
   ++ lib.optional uselibtirpc libtirpc;
 
   preConfigure = lib.optionalString uselibtirpc ''
@@ -66,7 +67,7 @@ stdenv.mkDerivation rec {
     substituteInPlace config/cmake/FindXDR.cmake \
       --replace 'find_path(XDR_INCLUDE_DIR NAMES rpc/types.h PATHS "/usr/include" "/usr/include/tirpc")' \
                 'find_path(XDR_INCLUDE_DIR NAMES rpc/types.h PATH_SUFFIXES include/tirpc)'
-  '' + lib.optionalString (szip != null) ''
+  '' + lib.optionalString szipSupport ''
     export SZIP_INSTALL=${szip}
   '';
 
@@ -85,7 +86,7 @@ stdenv.mkDerivation rec {
     "-DJAVA_HOME=${jdk}"
     "-DJAVA_AWT_LIBRARY=${javabase}/libawt.so"
     "-DJAVA_JVM_LIBRARY=${javabase}/server/libjvm.so"
-  ] ++ lib.optionals (szip != null) [
+  ] ++ lib.optionals szipSupport [
     "-DHDF4_ENABLE_SZIP_ENCODING=ON"
     "-DHDF4_ENABLE_SZIP_SUPPORT=ON"
   ];
@@ -118,6 +119,17 @@ stdenv.mkDerivation rec {
   postInstall = ''
     moveToOutput bin "$bin"
   '';
+
+  passthru = {
+    inherit
+      uselibtirpc
+      libtirpc
+      szipSupport
+      szip
+      javaSupport
+      jdk
+    ;
+  };
 
   meta = with lib; {
     description = "Data model, library, and file format for storing and managing data";

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -5900,7 +5900,9 @@ with pkgs;
 
   pixz = callPackage ../tools/compression/pixz { };
 
-  plplot = callPackage ../development/libraries/plplot { };
+  plplot = callPackage ../development/libraries/plplot {
+    inherit (xorg) libX11;
+  };
 
   pxattr = callPackage ../tools/archivers/pxattr { };
 

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -5969,9 +5969,7 @@ with pkgs;
 
   hddtemp = callPackage ../tools/misc/hddtemp { };
 
-  hdf4 = callPackage ../tools/misc/hdf4 {
-    szip = null;
-  };
+  hdf4 = callPackage ../tools/misc/hdf4 { };
 
   hdf5 = callPackage ../tools/misc/hdf5 {
     gfortran = null;

